src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -425,14 +426,16 @@ private function getOrderProducts(Order $order): OrderProductsForViewing
                 $product['unit_price_tax_excl'] :
                 $product['unit_price_tax_incl']
             ;
-            $totalPrice = Tools::ps_round($unitPrice, 2) * ($product['product_quantity'] - $product['customizationQuantityTotal']);
+            $totalPrice = Tools::ps_round($unitPrice, Context::getContext()->getComputingPrecision()) * ($product['product_quantity'] - $product['customizationQuantityTotal']);
 
             $unitPriceFormatted = $this->locale->formatPrice($unitPrice, $currency->iso_code);
             $totalPriceFormatted = $this->locale->formatPrice($totalPrice, $currency->iso_code);
 
             $imagePath = isset($product['image_tag']) ?
                 $this->imageTagSourceParser->parse($product['image_tag']) :
                 null;
+            $product['product_quantity_refunded'] = $product['product_quantity_refunded'] ?: false;
+            $product['amount_refund'] = $product['amount_refund'] ?: 0;
```

Comment:
Since it's supposed to be a formatted price, should you use `$this->locale->formatPrice(0, $currency->iso_code);` as a fallback? Unless the field should be nullable?
Besides why reset this array field once again when it's already checked L363? These ternary checks should be used for the OrderProductForViewing  constructor, not modify the previously formatted array. Or you can correct the code L363 as well

Reply:
done !